packages/Adapters/STKMesh/example/mesh_deformation.cpp

```diff
@@ -0,0 +1,396 @@
+                              stk::mesh::Selector &moving_selector,
+                              stk::mesh::Selector &interior_selector,
+                              Teuchos::Array<double> &coord_field,
+                              Teuchos::Array<double> &displace_field, double ux,
+                              double uy, int &num_boundary, int &num_interior )
+{
+    // Get the fixed boundary nodes.
+    stk::mesh::BucketVector fixed_part_buckets =
+        fixed_selector.get_buckets( stk::topology::NODE_RANK );
+    std::vector<stk::mesh::Entity> fixed_part_nodes;
+    stk::mesh::get_selected_entities( fixed_selector, fixed_part_buckets,
+                                      fixed_part_nodes );
+    int fixed_num_part_nodes = fixed_part_nodes.size();
+
+    // Get the moving boundary nodes.
+    stk::mesh::BucketVector moving_part_buckets =
+        moving_selector.get_buckets( stk::topology::NODE_RANK );
+    std::vector<stk::mesh::Entity> moving_part_nodes;
+    stk::mesh::get_selected_entities( moving_selector, moving_part_buckets,
+                                      moving_part_nodes );
+    int moving_num_part_nodes = moving_part_nodes.size();
+
+    // Get the number of boundary nodes.
+    num_boundary = fixed_num_part_nodes + moving_num_part_nodes;
+
+    // Get the interior nodes.
+    stk::mesh::BucketVector interior_part_buckets =
+        interior_selector.get_buckets( stk::topology::NODE_RANK );
+    std::vector<stk::mesh::Entity> interior_part_nodes;
+    stk::mesh::get_selected_entities( interior_selector, interior_part_buckets,
+                                      interior_part_nodes );
+    num_interior = interior_part_nodes.size();
+
+    // Allocate
+    int total_num_nodes = num_boundary + num_interior;
+    coord_field.resize( 3 * total_num_nodes );
+    displace_field.resize( 3 * total_num_nodes );
+
+    // Get the coordinate field.
+    const stk::mesh::FieldBase *coord_data_base =
+        bulk_data.mesh_meta_data().coordinate_field();
+    const stk::mesh::Field<double, stk::mesh::Cartesian3d> *coord_data =
+        dynamic_cast<const stk::mesh::Field<double, stk::mesh::Cartesian3d> *>(
+            coord_data_base );
+    double *coords;
+
+    // Create an offset.  f is an offset here. its used to index into a global
+    // array. The coordinate array is broken up into chunks of the fixed
+    // boundary, moving boundary, and interior node. The coordinates and
+    // displacements for these exist in different STK mesh parts so we break
+    // the coordinate and field assignment into loops over nodes in each part.
+    int f = 0;
+    
+    // Get the fixed data.
+    for ( int n = 0; n < fixed_num_part_nodes; ++n )
+    {
+        f = n;
+
+        displace_field[f * 3] = 0.0;
+        displace_field[f * 3 + 1] = 0.0;
+
+        coords = stk::mesh::field_data( *coord_data, fixed_part_nodes[n] );
+        coord_field[f * 3] = coords[0];
+        coord_field[f * 3 + 1] = coords[1];
```

Comment:
use `xxx_field[f * 2 + 0]` and `xxx_field[f * 2 + 1]`.  this is misleading, I was wondering "what about z-direction?"